### DISCUSSION TITLE
INFOREQ: PascalCase in namespaces

### DISCUSSION THREAD

**Discussion by ThorstenSuckow** [neomjs/neo#3199] [General]
As a follow up to https://github.com/neomjs/neo/pull/3193, I wanted to understand the reasoning behind the usage of PascalCase instead of camelCase with namespace parts, since this is something you rarely see nowadays. @tobiu can you shed some light on this?

**Comment by tobiu** [comment]
As mentioned inside Slack, it is mostly PascalCase for namespaces and class names, the rest is lowercase.

examples:
1. Covid.view.country.Helix
2. Neo.data.RecordFactory
3. Neo.main.addon.LocalStorage
4. Neo.table.Container
5. RealWorld.view.user.ProfileComponent

Makes sense or would you do it differently?

**Comment by ThorstenSuckow** [comment]
Okay, got it. Some companies like to use abbreviations for their app names (first namespace part) and I'm more used to `camelCase` all along except for `PascalCase` for `class names`. Does it make sense to make first and last part of the `fqn` `PascalCase`? I guess this is just personal taste and does not affect functionality though:

```java
import com.domain.namedPackage.MyClass;
```

vs.

```
LoginHelloWorld.view.LoginFormContainer
```
as a valid example for current conventions - although the app name might be chosen badly - leaves me visually confused... 👀 

Is the `PascalCase` for the app-name a **SHOULD BE** or a **MUST BE**?

**Comment by tobiu** [comment]
Hi Torsten,

PascalCase for app names is optional. Neo is storing app names inside a json file, since the build programs will make the app folder lowercase => we can figure out, how the internal namespace should be. Of course you could rename the folder, in case you want to use camelCase or whatever.

Inside the framework, I am using lowercase for all folder names, but this is optional as well. From a consistency perspective, we should keep it this way (or create a new concept and change it everywhere). inside the app / workspace scope, you can name folders however you like to.

I definitely agree that our workshop demo app name is not a good choice @davhm. Way too long.

tl-br: should be and NOT must be.
